contracts/ERC5018ForBlob.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IERC5018ForBlob.sol";
+
+interface EthStorageContract {
+    function putBlob(bytes32 key, uint256 blobIdx, uint256 length) external payable;
+
+    function get(bytes32 key, uint256 off, uint256 len) external view returns (bytes memory);
+
+    function remove(bytes32 key) external;
+
+    function size(bytes32 key) external view returns (uint256);
+
+    function hash(bytes32 key) external view returns (bytes24);
+
+    function upfrontPayment() external view returns (uint256);
+}
+
+contract ERC5018ForBlob is IERC5018ForBlob {
+
+    address public owner;
+    EthStorageContract public storageContract;
+
+    mapping(bytes32 => bytes32[]) internal keyToChunk;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function setEthStorageContract(address storageAddress) public {
+        require(msg.sender == owner, "must from owner");
+        storageContract = EthStorageContract(storageAddress);
+    }
+
+    function changeOwner(address newOwner) public {
+        require(msg.sender == owner, "must from owner");
+        owner = newOwner;
+    }
+
+    function _countChunks(bytes32 key) internal view returns (uint256) {
+        return keyToChunk[key].length;
+    }
+
+    function _chunkSize(bytes32 key, uint256 chunkId) internal view returns (uint256, bool) {
+        if (chunkId >= _countChunks(key)) {
+            return (0, false);
```

Comment:
can we just require it?

Reply:
The view function should not use 'require', because the 'require' front-end request will throw an exception.

Reply:
got it